docker/runtime/dotnetcore-2.0/Controllers/FunctionController.cs

```diff
@@ -0,0 +1,57 @@
+﻿using System;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using kubeless_netcore_runtime.Util;
+
+namespace kubeless_netcore_runtime.Controllers
+{
+    //[Produces("application/json")]
+    [Route("/")]
+    public class FunctionController : Controller
+    {
+
+        [HttpPost]
+        public object Post([FromBody]object data)
+        {
+            try
+            {
+                return CompileAndExecuteFunction(data);
+            }
+            catch (Exception ex)
+            {
+                return ex.Message;
+            }
+        }
+
+        [HttpGet]
+        public string Get()
+        {
+            return "NotImplemented";
+        }
+
+        private object CompileAndExecuteFunction(object data)
+        {
+            var className = Environment.GetEnvironmentVariable("MOD_NAME");
+            Console.WriteLine($"==> Class name: {className}");
+            var functionName = Environment.GetEnvironmentVariable("FUNC_HANDLER");
+            Console.WriteLine($"==> Function name: {functionName}");
+
+            var codeFile = string.Concat("/kubeless/", className, ".cs");
+            Console.WriteLine($"==> Code file: {codeFile}");
+            var code = System.IO.File.ReadAllText(codeFile);
+            Console.WriteLine($"==> Code:\n{code}");
+
+            var compiler = new Compiler(code, className, functionName);
+
+            //TODO: Install dependencies from project.json file
+            //var dependencies = "/kubeless/project.json";

```

Comment:
would this not be handled by the init container?

Reply:
When we first design the .NET Core runtime for kubeless, we thought it could be a great approach to compile it in runtime everytime, because the function could be easy changed in runtime. But actually your comment is really relevant. Maybe it would be better to compile it once during init - and just execute on the controller.
We are going to do some improvements on the runtime:
* Requirements handling
* Compilation at init time
* Implementation of GET method.

Reply:
Implemented (compilation and GET) :)